npm/webpack-preprocessor/lib/cross-origin-callback-store.ts

```diff
@@ -0,0 +1,29 @@
+export interface CrossOriginCallbackStoreFile {
+  inputFileName: string
+  outputFilePath: string
+  source: string
+}
+
+export class CrossOriginCallbackStore {
+  private files: { [key: string]: CrossOriginCallbackStoreFile[] } = {}
+
+  addFile (sourceFilePath: string, file: CrossOriginCallbackStoreFile) {
+    this.files[sourceFilePath] = (this.files[sourceFilePath] || []).concat(file)
```

Comment:
Just curious, with our current file watchers in the data-context, we are watching more than we were in 9.x in the `cypress` directory. If `Cypress.require()` is used on a file within the dir, is replacing those contents in this list or appending it even it it prev existed?

Reply:
This store is used to track files that will _eventually_ be created when the loader finds uses of `Cypress.require()`. It's completely in-memory. Nothing is written to disk until webpack runs through the files referenced and compiles them. It doesn't currently add any more watching, though that is something that will need to be implemented. Even then, we won't need to worry about it since webpack will handle it for us.